packages/radix-ui-themes/src/components/radio.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import * as React from 'react';
+import classNames from 'classnames';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { composeRefs } from '@radix-ui/react-compose-refs';
+import { radioPropDefs } from './radio.props.js';
+import { extractProps } from '../helpers/index.js';
+import { marginPropDefs } from '../props/index.js';
+
+import type { ComponentPropsWithoutColor, NotInputRadioAttributes } from '../helpers/index.js';
+import type { MarginProps, GetPropDefTypes } from '../props/index.js';
+
+type RadioElement = React.ElementRef<'input'>;
+type RadioOwnProps = GetPropDefTypes<typeof radioPropDefs> & {
+  value: string;
+  onValueChange?: (value: string) => void;
+};
+type RadioInputProps = Omit<
+  ComponentPropsWithoutColor<'input'>,
+  NotInputRadioAttributes | 'defaultValue' | 'value'
+>;
+interface RadioProps extends RadioInputProps, MarginProps, RadioOwnProps {}
+
+const Radio = React.forwardRef<RadioElement, RadioProps>((props, forwardedRef) => {
+  const ref = React.useRef<RadioElement>(null);
+  const { className, color, onChange, onValueChange, ...radioProps } = extractProps(
+    props,
+    radioPropDefs,
+    marginPropDefs
+  );
+  return (
+    <input
+      type="radio"
```

Comment:
At first I attempted to implement this component extracting the internal Radio primitive, but outside the Radio Group it was underwhelming compared to the native radio input:
- Can be only controlled
- No automatic roving focus between radios with matching `name` attribute

Reply:
Agreed, not worth it.